internal/gojs/fs.go

```diff
@@ -325,10 +325,18 @@ func syscallReaddir(_ context.Context, mod api.Module, name string) (*objectArra
 	}
 	defer f.Close() //nolint
 
-	if dirents, errno := f.Readdir(-1); errno != 0 {
+	dirs, errno := f.Readdir()
+	if errno != 0 {
 		return nil, errno
-	} else {
-		entries := make([]interface{}, 0, len(dirents))
+	}
+
+	dirents, errno := fsapi.Collect(dirs)
```

Comment:
currently, it is also being used in `func (d *openRootDir) readdir() syscall.Errno`, not necessarily because it is right, but because the logic there seemed a bit articulated and I didn't want to touch too many places at once.

I can already update the tests to use a simplified version that takes the file instead, though.